### PR TITLE
Add spinner while waiting for form pages to load

### PR DESCRIPTION
**Pull request by andrewadcock**
Created a spinner component and styled. 

The component is added to WrapSecurity. This is initiated via dispatch in `intial.js`, shown only on `/sections/` pages, and is terminated in `Part.js`.

### CODE REVIEW

**Review by cl-adavyd — APPROVED** [review]

**Review by mblevins-collabralink — APPROVED** [review]

File: frontend/react/src/components/layout/Part.js [hunk 1]
```diff
@@ -91,7 +91,13 @@ const mapStateToProps = (state, { partId }) => {
...
   };
 };
 
-export default connect(mapStateToProps)(Part);
+const mapDispatchToProps = (dispatch) => {
+  // End spinner, started in initial.js
+  dispatch({ type: "CONTENT_FETCHING_FINISHED" });
+};
+
+export default connect(mapStateToProps, mapDispatchToProps)(Part);
```
Comment (mgwalker) [inline frontend/react/src/components/layout/Part.js:103]: I think using `mapDispatchToProps` for side-effects would be an anti-pattern. A future dev wouldn't know to look here for anything other than mapping actions to component properties. I get why you went this route, but there's another way! 😄

File: frontend/react/src/actions/initial.js [hunk 2]
```diff
@@ -129,6 +129,9 @@ export const loadForm = (state) => async (dispatch, getState) => {
   const { stateUser } = getState();
   const stateCode = state ?? stateUser.currentUser.state.id;
 
+  // Start isFetching for spinner; turned off in Part.js
+  dispatch({ type: "CONTENT_FETCHING_STARTED" });
+
   dispatch(loadSections({ userData: stateUser, stateCode }));
   dispatch(getStateStatus({ stateCode }));
   dispatch(getAllStatesData());
```
Comment (mgwalker) [inline frontend/react/src/actions/initial.js:137]: All of these dispatches return promises, so you can do something like:

```js
dispatch({ type: "CONTENT_FETCHING_STARTED" });

await Promise.all([
  dispatch(loadSections({ userData: stateUser, stateCode })),
  dispatch(getStateStatus({ stateCode })),
  dispatch(getAllStatesData()),
]);

dispatch({ type: "CONTENT_FETCHING_FINISHED" });
```

**Review by mgwalker — COMMENTED** [review]
TIL an animated gif will keep animating even when the UI thread is blocked. Nice!

File: frontend/react/src/actions/initial.js (hunk 2 above)
Comment (andrewadcock) [reply frontend/react/src/actions/initial.js]: Thank you! That's much cleaner and allowed me to remove the code from part.js and also the check for what pages show the spinner. Thanks!

**Review by mgwalker — APPROVED** [review]

**Pull request merged**